engine/src/test/java/io/zeebe/engine/processing/incident/CallActivityIncidentTest.java

```diff
@@ -31,40 +34,50 @@
 
   @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
 
-  private static final String PROCESS_ID_PARENT = "wf-parent";
-  private static final String PROCESS_ID_CHILD = "wf-child";
-
   private static final String PROCESS_ID_VARIABLE = "wfChild";
 
-  private static final BpmnModelInstance WORKFLOW_PARENT =
-      Bpmn.createExecutableProcess(PROCESS_ID_PARENT)
-          .startEvent()
-          .callActivity("call", c -> c.zeebeProcessId(PROCESS_ID_CHILD))
-          .done();
-
-  private static final BpmnModelInstance WORKFLOW_PARENT_PROCESS_ID_EXPRESSION =
-      Bpmn.createExecutableProcess(PROCESS_ID_PARENT)
-          .startEvent()
-          .callActivity("call", c -> c.zeebeProcessIdExpression(PROCESS_ID_VARIABLE))
-          .done();
-
-  private static final BpmnModelInstance WORKFLOW_CHILD =
-      Bpmn.createExecutableProcess(PROCESS_ID_CHILD).startEvent().endEvent().done();
+  private static final BiFunction<String, String, BpmnModelInstance> WORKFLOW_PARENT_SUPPLIER =
```

Comment:
How about just inlining this code? We only call the suppliers in one place anyway, so it's not like we reuse them much, and we don't save from not rebuilding the process anymore anyway.

Reply:
I had this as my first iteration, but then I thought it would be more save in case we want to add more tests to be explicit about using suppliers, so we don't forget about it the next time we write a test in this class.

Reply:
I'm not sure I understand - it's so when I come to this class, and I forgot about this specific issue of reused process IDs, seeing the static suppliers will highlight it? I'm not sure about it, but like I said, it's not a big deal, and time will tell if that's correct or not :smile: I would propose just adding a note to the class's javadoc about this if we really want to make sure to remember (e.g. `Note that tests in this class are sensitive to reused process IDs, so make sure to generate one per case - see {@link #setUp()} as an example`).

Reply:
Not sure about this node as this should be a general best practice for all test which use the workflow engine `ClassRule`. I will inline this code as it removes from lines, and then we will deal with it once we break it again.

Reply:
You're right, it's something we should be careful everywhere we use the class rule. Not sure though how we can prevent this everywhere :thinking: